src/utilities/theme/utils.ts

```diff
@@ -366,3 +387,9 @@ function rem(px: string) {
   const baseFontSize = 10;
   return `${parseInt(px, 10) / baseFontSize}rem`;
 }
+
+function buildBannerBorder(cssVar: string) {
+  return `inset 0 ${rem('2px')} 0 0 var(${cssVar}), inset 0 0 0 ${rem(
```

Comment:
I believe you can nest a css custom property inside another, that might negate the need for this js helper.


Something like the following - written but not tested:

```css
.Banner {
  --p-banner-border: inset 0 rem(2px) 0 0 var(--p-banner-color), inset 0 0 0 rem(2px) var(--p-banner-color)
}

.statusSuccess {
  /* plus some indirection with those mixins */  
  --p-banner-color: var(--p-success-divider, color('green') )
}
```

Reply:
That approach won’t work because the border style (actually a box shadow) is different between old and new banners. So we need the the entire new box shadow value to be fully defined in JS or undefined:

```scss
.Banner {
  --p-banner-border: inset 0 rem(2px) 0 0 var(--p-banner-color), inset 0 0 0 rem(2px) var(--p-banner-color)
}

.statusSuccess {
  /* plus some indirection with those mixins */  
  --p-banner-color: var(--p-success-divider, color('green') )

  // this doesn’t work as we never fallback to the old value
  box-shadow: var(--p-banner-border, OLD_VALUE_HERE);
}
```

For helpers, we still need the `rem` function for border radius and probably will for other values. The border helper is for readability and to dry up the code a bit.

Reply:
Cool, thanks for the context. Roll on the time when all this is done and we can simplify these forked cases :D